operatorcourier/validate.py

```diff
@@ -85,30 +93,27 @@ def _bundle_validation(self, bundle):
 
     def _crd_validation(self, bundleData):
         logger.info("Validating custom resource definitions.")
-        valid = True
-
         crds = bundleData[self.crdKey]
 
+        valid = True
         for crd in crds:
-            if "metadata" in crd:
-                if "name" in crd["metadata"]:
-                    logger.info("Evaluating crd %s", crd["metadata"]["name"])
-                else:
-                    self._log_error("crd metadata.name not defined.")
-                    valid = False
-            else:
-                self._log_error("crd metadata not defined.")
-                valid = False
-
-            if "apiVersion" not in crd:
-                self._log_error("crd apiVersion not defined.")
-                valid = False
+            for crd_required_nested_field in self.crd_required_nested_fields:
+                valid &= self._crd_fields_validation(crd_required_nested_field.split('.'),
+                                                     crd, "")
+        return valid
 
-            if "spec" not in crd:
-                self._log_error("crd spec not defined.")
-                valid = False
+    def _crd_fields_validation(self, fields, yaml_dict, prefix):
+        if len(fields) == 0:
+            return True
+        root_field, subfields = fields[0], fields[1:]
+        field_string = root_field if prefix == "" else prefix + "." + root_field
 
-        return valid
+        if root_field == 'metadata' and 'name' in yaml_dict['metadata']:
+            logging.info("Evaluating crd %s", yaml_dict['metadata']['name'])
+        if root_field not in yaml_dict:
+            self._log_error("crd %s not defined.", field_string)
```

Comment:
nit: Can this also print the name of the crd in question? Or is that covered by previous if statement?

Reply:
> Or is that covered by previous if statement?

Yes.

> Can this also print the name of the crd in question?

This actually prints out the (nested) field that is missing in the CRD. See this [test case](https://github.com/operator-framework/operator-courier/pull/41/files#diff-7488fbee800ac1a3edef38842281bda6R66) as an example. I have updated the log message to make it less confusing.